File: rximagepicker/src/main/java/com/qingmei2/rximagepicker/entity/CustomPickConfigurations.java
package com.qingmei2.rximagepicker.entity;

import com.qingmei2.rximagepicker.ui.ICustomPickerConfiguration;

import java.util.Map;

public final class CustomPickConfigurations {

    private Map<String, ICustomPickerConfiguration> configurations;

    public CustomPickConfigurations(Map<String, ICustomPickerConfiguration> configurations) {
        if (configurations == null)
            throw new NullPointerException("configurations is null.");
        this.configurations = configurations;
    }

    public void resetConfigrations(Map<String, ICustomPickerConfiguration> configurations) {
        this.configurations = configurations;
    }

    public ICustomPickerConfiguration findConfigurationByKey(String viewKey) {
        if (viewKey == null)
            throw new NullPointerException("viewKey is null");
        return configurations.get(viewKey);
    }
}

File: rximagepicker/src/main/java/com/qingmei2/rximagepicker/entity/CustomPickConfigurations.kt
package com.qingmei2.rximagepicker.entity

import com.qingmei2.rximagepicker.ui.ICustomPickerConfiguration

class CustomPickConfigurations(private var configurations: Map<String, ICustomPickerConfiguration>) {

    fun resetConfigrations(configurations: Map<String, ICustomPickerConfiguration>) {
        this.configurations = configurations
    }

    fun findConfigurationByKey(viewKey: String): ICustomPickerConfiguration {
        return configurations[viewKey]!!
    }
}
